modules/zip/src/parse-zip/end-of-central-directory.ts

```diff
@@ -56,10 +65,84 @@ export const parseEoCDRecord = async (file: FileProvider): Promise<ZipEoCDRecord
 
     cdRecordsNumber = await file.getBigUint64(zip64EoCDOffset + ZIP64_CD_RECORDS_NUMBER_OFFSET);
     cdStartOffset = await file.getBigUint64(zip64EoCDOffset + ZIP64_CD_START_OFFSET_OFFSET);
+  } else {
+    zip64EoCDLocatorOffset = 0n;
   }
 
   return {
     cdRecordsNumber,
-    cdStartOffset
+    cdStartOffset,
+    offsets: {
+      zip64EoCDOffset,
+      zip64EoCDLocatorOffset,
+      zipEoCDOffset
+    }
   };
 };
+
+/**
+ * updates EoCD record to add one more file to the archieve
+ * @param eocdBody buffer containing header
+ * @param oldEoCDinfo info read from EoCD record befor updating
+ * @param newCDStartOffset CD start offset to be updated
+ * @param eocdStartOffset EoCD start offset to be updated
+ * @returns new EoCD header
+ */
+export async function updateEoCD(
```

Comment:
If we add a lot of files, there seems to be a lot of repeated copying of a bigger and bigger record again and again. This is often a perf issue. 

In these cases it is often better to build up the list of things you want in the buffer and then at the end generate it.

Might not be worth the effort now, but something to think about.

Reply:
This function is only about End of Central Directory record and EoCD isn't getting bigger with more files in the archive, we're only increasing numbers there. But anyway, I'll improve it to work with addition of many files.